### ISSUE TITLE
Permission To Allow Operation Based On Ownership

### ISSUE THREAD

**Issue by imkrishnaagrawal** [amplication/amplication#3974]
### Feature description

I was thinking the permission section of the app . just like we have actions based on roles we can also add more control based on the conditions. if particular attribute of that field match to some value then we can allow it to perform action like View or Create as an generic feature.
One Of The use case I could think of is allow owner of entity to update not all user with same role.

### Use case

I have created an application with few roles and entities , while configuring the permission I wanted to allow certain user like Owner to update the entity however that's not an option. I know I can do it myself by making changes in the code. that's not a blocker however I think the feature will add lot of value to me as consumer of amplication and others out there.

**Comment by souravjain540** [comment]
Thank you for your suggestion @imkrishnaagrawal ! One of the team members will review this very soon.

**Comment by souravjain540** [comment]
@arielweinberger  your views?

**Comment by NisanAyash** [comment]
That's a good idea,
It may be helpful to consider `RBAC`.

**Comment by arielweinberger** [comment]
Thanks for the suggestion. It's a pretty busy holiday season for us so we'll make sure to get back to you ASAP.

@GreenMachine01 and @yuval-hazaz will be able to help as they steer the roadmap.

@NisanAyash We already have RBAC in place. I assume you mean ABAC? (Attribute Based Access Control)

**Comment by imkrishnaagrawal** [comment]
Yes @arielweinberger I do agree with you and I understand amount of efforts you guys have to spend for the feature. I have build similar application using CASL , NestJs and TypeOrm where I have used AWS like policies stored in database and reflect metadata to pass permission between components which can be attached to roles and I can control the permissions using those policies. I have also designed it to give permission to access and perform operations on certain fields of tables based on conditions. I have also added features to standardize the error responses and data responses, caching , read replicas support and more.  my template is similar to amplication however I dont have UI to manage and create different component. I found the project really intresting. I'll most probably use it for my other projects.  I would be happy to contribute to the project if required.

**Comment by clayrisser** [comment]
A simple initial approach to this could simply allow you to select a field that contains the owner id. For example, a user could create a relationship of a todo entity with a user using field such as owner. Then it would be trivial for the system to know the owner.

**Comment by yuval-hazaz** [comment]
@GreenMachine01 this issue should be considered as part of https://github.com/amplication/amplication/issues/4503

**Comment by zdanl** [comment]
The ownership part is very important, and I think #4503 articulates it well, but granularity in operations specific to the business logic of an application I would imagine could be realized neatly and efficiently by storing an integer for each user and working the privileges with bitfields.

https://github.com/retroscepter/bitfields
